custom/enikshay/integrations/bets/repeaters.py

```diff
@@ -51,12 +51,146 @@ def allowed_to_forward(self, voucher_case):
         return (
             approved
             and not_sent
+            # TODO: Check if voucher is related to test location?
             and case_properties_changed(voucher_case, ['state'])
         )
 
 
+class BETS180TreatmentRepeater(BaseBETSRepeater):
+    friendly_name = _("BETS - MBBS+ Providers: 180 days of private OR govt. FDCs with treatment outcome reported (episode case type)")
+
+    @classmethod
+    def get_custom_url(cls, domain):
+        from custom.enikshay.integrations.bets.views import BETS180TreatmentRepeaterView
+        return reverse(BETS180TreatmentRepeaterView.urlname, args=[domain])
+
+    def allowed_to_forward(self, episode_case):
+        if not self.case_types_and_users_allowed(episode_case):
+            return False
+
+        case_properties = episode_case.dynamic_case_properties()
+        treatment_outcome = case_properties.get('treatment_outcome', None)
+        episode_has_outcome = treatment_outcome and (treatment_outcome != 'not_evaluated')
+        adherence_total_doses_taken = case_properties.get('adherence_total_doses_taken', "0")
+        try:
+            adherence_total_doses_taken = int(adherence_total_doses_taken)
+        except ValueError:
+            adherence_total_doses_taken = 0
+
+        not_sent = case_properties.get("event_{}".format(TREATMENT_180_EVENT)) != "sent"
+        return (
+            episode_has_outcome
+            and not_sent
+            and adherence_total_doses_taken >= 180
+        )
+
+
+class BETSDrugRefillRepeater(BaseBETSRepeater):
+    friendly_name = _("BETS - Patients: Cash transfer on subsequent drug refill (voucher case type)")
+
+    @classmethod
+    def get_custom_url(cls, domain):
+        from custom.enikshay.integrations.bets.views import BETSDrugRefillRepeaterView
+        return reverse(BETSDrugRefillRepeaterView.urlname, args=[domain])
+
+    def allowed_to_forward(self, voucher_case):
+        if not self.case_types_and_users_allowed(voucher_case):
+            return False
+
+        voucher_case_properties = voucher_case.dynamic_case_properties()
+        episode = get_episode_case_from_voucher(voucher_case.domain, voucher_case.case_id)
+        episode_case_properties = episode.dynamic_case_properties()
+
+        voucher_count = episode_case_properties.get('approved_voucher_count', 0)
+        if voucher_count < 2:
+            voucher_count = len(get_approved_prescription_vouchers_from_episode(episode.domain, episode.case_id))
+
+        not_sent = voucher_case_properties.get("event_{}".format(DRUG_REFILL_EVENT)) != "sent"
+        return (
+            voucher_count >= 2
```

Comment:
can you make this one of the last checks? Since it potentially does a bunch of db lookups, we would only want it to trigger if all the other cases hold true.